### DISCUSSION TITLE
Solidity Version Errors

### DISCUSSION THREAD

**Discussion by jacksanford1** [tinchoabbate/damn-vulnerable-defi#59] [Help / Q&A / Troubleshooting] [answered]
Hi,

I'm trying to initialize the Damn Vulnerable DeFi project using `forge build`. I have Foundry installed so I'm not using the provided devcontainer for setup. 

I get an error noting that some of the test files are solidity version `=0.8.25` but some of the imports for those test files are solidity version `^0.8.4`.

For example, `Withdrawal.t.sol` is solidity version `=0.8.25` but it imports L1Forwarder (`../../src/withdrawal/L1Forwarder.sol`) which then imports Ownable (`solady/auth/Ownable.sol`) and Ownable is solidity version `=0.8.25`. 

I checked to see if I could import an earlier version of Ownable but it seems the file [started out](https://github.com/Vectorized/solady/commit/4634424963caf6abb5fcf97f5731c3d708af0144#diff-bd6786a626cf4e45c8e1dd3cf488dff2e220414e9c30280032d27c5cf88f6d06) with version `=0.8.25`. 

What is the right way to fix this?

Thanks. 

<img width="642" alt="image" src="https://github.com/user-attachments/assets/42b1bbc4-40ea-4e46-8b50-d784c1ff6695">

**Comment by tinchoabbate** [comment]
Hi @jacksanford1, would you mind running `forge --version` and sharing the output here? Might be useful to reproduce.

**Reply by tinchoabbate** [reply]
I just tested with a clean environment using `forge 0.2.0 (fdfaafd 2024-07-30T00:19:45.054521589Z)` and the compilation seems to work ok. I ran:

1. `git clone git@github.com:theredguild/damn-vulnerable-defi.git`
2. `git checkout v4.0.0`
3. `forge build`

And got:

```
[⠊] Compiling...
[⠒] Compiling 213 files with Solc 0.8.25
[⠘] Installing Solc version 0.8.25
[⠰] Successfully installed Solc 0.8.25
[⠒] Solc 0.8.25 finished in 12.05s
Compiler run successful!
``

**Reply by jacksanford1** [reply, marked as answer]
Thank you 🙏 I was indeed using an old forge version:

`forge 0.2.0 (a0db055 2022-04-03T00:03:53.441110+00:00)`

And didn't notice I was running into [this error](https://github.com/foundry-rs/foundry/issues/6152) when trying to update Foundry.

Using the latest forge version allows it to compile successfully 👍
